sdk/cosmosdb/cosmos/src/CosmosClient.ts

```diff
@@ -93,6 +94,9 @@ export class CosmosClient {
       async (opts: RequestOptions) => this.getDatabaseAccount(opts)
     );
     this.clientContext = new ClientContext(optionsOrConnectionString, globalEndpointManager);
+    if (optionsOrConnectionString.connectionPolicy?.enableEndpointDiscovery) {
```

Comment:
we should make sure the ref docs for `enableEndpointDiscovery` talk about the requirement to call `dispose`